### PR TITLE
Fix ASAN errors in API tests [15965]

### PR DESCRIPTION
**Pull request by jparisu**
## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [ ] Documentation builds and tests pass locally. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by jsantiago-eProsima** [comment]
@richiprosima please test this

File: test/blackbox/common/BlackboxTests.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ class BlackboxEnvironment : public ::testing::Environment
     {
         //Log::Reset();
         eprosima::fastdds::dds::Log::KillThread();
-        eprosima::fastrtps::rtps::RTPSDomain::stopAll();
+        // eprosima::fastrtps::rtps::RTPSDomain::stopAll();
```
Comment (rsanchez15) [inline test/blackbox/common/BlackboxTests.cpp:82]: Remove this

File: test/unittest/dds/participant/ParticipantTests.cpp [hunk 2]
```diff
@@ -274,6 +274,17 @@ class BarType
     std::array<char, 256> message_;
 };
 
+int process_id()
```
Comment (rsanchez15) [inline test/unittest/dds/participant/ParticipantTests.cpp:277]: I'd add a brief comment here on why `SystemInfo.cpp` cannot be included here.

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Comment by rsanchez15** [comment]
@richiprosima Please test this

**Comment by rsanchez15** [comment]
@richiprosima Please test this

File: test/unittest/dds/participant/CMakeLists.txt [hunk 3]
```diff
@@ -17,9 +17,8 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/../profiles/test_xml_profiles.xml
     COPYONLY)
 
 set(PARTICIPANTTESTS_SOURCE
-    ParticipantTests.cpp
-    ${PROJECT_SOURCE_DIR}/src/cpp/rtps/builtin/discovery/participant/ServerAttributes.cpp
```
Comment (rsanchez15) [inline test/unittest/dds/participant/CMakeLists.txt:21]: This makes the windows compilation fail as 
`template<> const char* server_ostream_separators<char>::locator_separator = "|";` 
is not exported.

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: test/unittest/dds/participant/CMakeLists.txt (hunk 3 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/dds/participant/CMakeLists.txt:21]: Also, linters are failing (uncrustify).

Kudos for resolving the fastbins memory corruption :partying_face: and I am still trying to stop laughing about the :crying_cat_face: poor kitten crying :laughing:

**Comment by jparisu** [comment]
@richiprosima Please test this

**Comment by jparisu** [comment]
@richiprosima Please test this

**Comment by jparisu** [comment]
@richiprosima Please test this

**Comment by jparisu** [comment]
@richiprosima Please test this

**Comment by jparisu** [comment]
@richiprosima Please test this

**Comment by jparisu** [comment]
@richiprosima Please test this

**Comment by jparisu** [comment]
**ATTENTION**: uncrustify failure must not be taken into account, as the proposed fix is a gramatical error.

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
CI failures are unrelated. Thanks @jparisu!

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**